.github/workflows/release.yml

```diff
@@ -1,175 +1,109 @@
-name: Build Release
+name: Release
 on:
   push:
     tags:
-    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
 
 jobs:
   build:
-    name: Build Release Artifacts
-    runs-on: ${{ matrix.os }}
-
+    name: Build Binary for ${{ matrix.target }}
     strategy:
-      fail-fast: true
+      fail-fast: false
       matrix:
-        os:
-        - macos-latest
-        - ubuntu-latest
-        - windows-latest
-
+        target:
+          - x86_64-unknown-linux-gnu
+          - x86_64-unknown-linux-musl
+          - x86_64-apple-darwin
+          - aarch64-apple-darwin
+          - x86_64-pc-windows-msvc
+        # This expands the matrix variables for the target
+        # https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-20.04
+            name: x86_64-unknown-linux-gnu.tar.gz
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-latest
+            name: x86_64-unknown-linux-musl.tar.gz
+          - target: x86_64-apple-darwin
+            os: macOS-latest
+            name: x86_64-apple-darwin.tar.gz
+          - target: aarch64-apple-darwin
+            os: macOS-latest
+            name: aarch64-apple-darwin.tar.gz
+          - target: x86_64-pc-windows-msvc
+            os: windows-latest
+            name: x86_64-pc-windows-msvc.zip
+    runs-on: ${{ matrix.os }}
     env:
```

Comment:
You could consider putting this as the top-level so that it applies to both jobs.

I also normally include `CARGO_TERM_COLOR: always` to ensure you get color output: https://github.com/google/comprehensive-rust/blob/main/.github/workflows/build.yml#L10.